src/rebar3_hex_client.erl

```diff
@@ -96,8 +96,8 @@ delete_docs(Config, Name, Version) ->
     response(Res).
 
 retire(Config, Package, Version, Reason, Message) -> 
-    Msg = #{<<"reason">> => Reason,
-            <<"message">> => Message
+    Msg = #{reason => Reason,
+            message => Message
```

Comment:
Maybe this wasn't throwing an exception / Dialyzer issue because somehow the keys are coerced into atoms/binaries remotely, but this is the way the spec is telling it should be.

Reply:
It doesn't work for some reason, I did an integration test locally using hexpm from main. 

Calling on @wojtekmach for clarity, maybe I was holding it wrong? I can try again.

Reply:
Ah, then the problem might be that the specs are incorrect, in which case the code was Ok. (and not under some coercion as I'd initially thought)

Reply:
@starbelly, might this also indicate we're missing an important test somewhere?

Reply:
@paulo-ferraz-oliveira We'd have to have a full blown integration test, which is doable.

Reply:
I'm Ok with identifying it as an issue and postponing it, though :wink

Reply:
Yeah, so I peaked further and it appears (as you noticed) that `hexpm` requires that as a string. The spec says `atom` which is incorrect.

I see several solutions, among which:

- drop the fine-grained spec (because you can't spec-ify a map's binary keys)
- accept atoms and convert them to binaries
- add a `-dialyzer` directive (dirtier) and move on

Reply:
Opened issue : https://github.com/hexpm/hex_core/issues/149

Reply:
Opened pr 😄: https://github.com/hexpm/hex_core/pull/150

Reply:
I think now the "bug" boils down to what's more important (and that's probably interpreted differently by different people):

* is it the spec that's Ok and we should adapt the code (as per the pull request)
* is it the code that's Ok and we should adapt the spec (in which case it'll be less readable)

I personally think it doesn't hurt to support both the binary `rebar3_hex` is sending while making the public intention the atom and then manipulating the input inside the lib.

Reply:
I think that would be ok. Similar to what Eric stated on hex_core discussion, it is a bit more to maintain. By no means a heavy lift ever, but still, there is that. 

I suppose atm, we're waiting for your PR to get merged and new release of hex_core to be cut.

Reply:
Yeah, I rebased over the main branch, and reversed this particular change (reason/message) so as soon as a new `hex_core` gets cut we can update here, then merge.

Reply:
Resolve'ing, as this brings nothing new.